### PR TITLE
TS-4729: Remove dead assaignment in Http2Stream

### PR DESCRIPTION
**Pull request by masaori335**
[TS-4729](https://issues.apache.org/jira/browse/TS-4729)

### CODE REVIEW

**Comment by zwoop** [comment]
I assume is_done is still in use elsewhere. 
👍

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/514/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/411/ for details.

File: proxy/http2/Http2Stream.cc [hunk 1]
```diff
@@ -509,7 +509,7 @@ Http2Stream::update_write_request(IOBufferReader *buf_reader, int64_t write_len,
         parent->connection_state.send_headers_frame(this);
 
         // See if the response is chunked.  Set up the dechunking logic if it is
-        is_done = this->response_initialize_data_handling();
+        this->response_initialize_data_handling();
```
Comment (jpeach) [inline proxy/http2/Http2Stream.cc]: Should you check or assert on the return value here? I would guess that bad things happen if ``response_initialize_data_handling`` returns false.

**Comment by jpeach** [comment]
Also, there's a typo in the commit subject line ;)

File: proxy/http2/Http2Stream.cc (hunk 1 above)
Comment (bryancall) [inline proxy/http2/Http2Stream.cc]: The boolean is not related to if `response_initialize_data_handling` was successful or not, which is bad coding in my opinion.  It is information about if the response was chunked and if it is finished.

The return value for `response_initialize_data_handling` should be changed.

Comment (shinrich) [inline proxy/http2/Http2Stream.cc]: Actually, the is_done's between response_intiailize_data_handling and response_process_data are related.  If the response is not chunked both will always return false.  If they are chunked, they both return true if the entire chunk was represent and could be read.

It looks like my patch to open source missed the check on the return from this->response_initialize_data_handling.  There should be

if (is_done) {
  send_event = VC_EVENT_WRITE_COMPLETE;
}

I'll add that and set up a new PR.

**Comment by masaori335** [comment]
#849 is the new one by @shinrich

**Pull request closed without merging**